lib/Element.js

```diff
@@ -1,986 +1,438 @@
-    get: getLastChildElement
-  });
-
-  Object.defineProperty(element, "previousElementSibling", {
-    get: () => _getElement($elm.prev())
-  });
-
-  Object.defineProperty(element, "nextElementSibling", {
-    get: () => _getElement($elm.next())
-  });
-
-  Object.defineProperty(element, "children", {
-    get: getChildren
-  });
-
-  Object.defineProperty(element, "childNodes", {
-    get: getChildNodes
-  });
-
-  Object.defineProperty(element, "parentElement", {
-    get: () => _getElement($elm.parent())
-  });
-
-  Object.defineProperty(element, "parentNode", {
-    get: () => _getElement($elm.parent())
-  });
-
-  Object.defineProperty(element, "innerHTML", {
-    get: () => $elm.html(),
-    set: (value) => {
-      $elm.html(value);
-      if (tagName === "textarea") {
-        element.value = $elm.html();
-      }
-
-      emitter.emit("_insert");
-    }
-  });
-
-  Object.defineProperty(element, "innerText", {
-    get: () => element.textContent,
-    set: (value) => {
-      element.textContent = value;
-
-      if (tagName === "textarea") {
-        element.value = element.textContent;
-      }
-    }
-  });
-
-  Object.defineProperty(element, "textContent", {
-    get: () => {
-      return tagName === "script" ? $elm.html() : $elm.text();
-    },
-    set: (value) => {
-      const response = tagName === "script" ? $elm.html(value) : $elm.text(value);
-      emitter.emit("_insert");
-      return response;
-    }
-  });
-
-  Object.defineProperty(element, "outerHTML", {
-    get: () => {
-      return $.html($elm);
-    },
-    set: (value) => {
-      $elm.replaceWith($(value));
-      emitter.emit("_insert");
-    }
-  });
-
-  Object.defineProperty(element, "href", {
-    get: () => {
-      const rel = getAttribute("href");
-      return makeAbsolute(location, rel);
-    },
-    set: (value) => {
-      setAttribute("href", value);
-    }
-  });
-
-  Object.defineProperty(element, "src", {
-    get: () => {
-      const rel = getAttribute("src");
-      return makeAbsolute(location, rel);
-    },
-    set: (value) => {
-      setAttribute("src", value);
-      dispatchEvent(new Event("load", {bubbles: true}));
-    }
-  });
-
-  Object.defineProperty(element, "content", {
-    get: () => {
-      if (tagName !== "template") return;
-      return DocumentFragment(Element, element);
-    }
-  });
-
-  Object.defineProperty(element, "checked", {
-    get: () => getProperty("checked"),
-    set: (value) => {
-      if ($elm.attr("type") === "radio") radioButtonChecked(value);
-      else if ($elm.attr("type") === "checkbox") checkboxChecked(value);
-    }
-  });
+const inputElements = ["INPUT", "BUTTON", "TEXTAREA"];
+
+const classListSymbol = Symbol.for("classList");
+const datasetSymbol = Symbol.for("dataset");
+const rectsSymbol = Symbol.for("rects");
+const scrolledSymbol = Symbol.for("scrolled");
+const stylesSymbol = Symbol.for("styles");
+
+module.exports = class Element extends EventTarget {
```

Comment:
should class be split up into `Element` and `HTMLElement`? currently it's a mixture.